File: Assets/Scripts/Core/GameStartValidationResult.cs
namespace Bunker.Core
{
    // Result of pre-start checks, so the lobby UI can show a specific,
    // actionable message instead of a generic "cannot start" error.
    public class GameStartValidationResult
    {
        public bool CanStart;
        public string FailReason;

        public static GameStartValidationResult Ok() => new() { CanStart = true };
        public static GameStartValidationResult Fail(string reason) =>
            new() { CanStart = false, FailReason = reason };
    }
}

File: bunker-server/src/game/GameStartValidationResult.ts
// Equivalent of the C# GameStartValidationResult (section 1.7). Lets the
// lobby/room code show a specific, actionable message before attempting to
// start the game, instead of only failing deep inside startGame().
export interface GameStartValidationResult {
  canStart: boolean;
  failReason?: string;
}

export function validationOk(): GameStartValidationResult {
  return { canStart: true };
}

export function validationFail(reason: string): GameStartValidationResult {
  return { canStart: false, failReason: reason };
}
